bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/TelescopePosTableWidget.java

```diff
@@ -917,15 +917,20 @@ private void _resetTable(final TargetEnvironment newEnv) {
     void updatePrimaryStar() {
         if (_env == null || !OTOptions.isEditable(_obsComp.getProgram(), _obsComp.getContextObservation())) return;
 
-        // TODO: This method must be tested in greater detail, since custom guiding is required for offset positions.
+        final TargetEnvironment env = _dataObject.getTargetEnvironment();
         final Option<SPTarget> targetOpt = getSelectedPos();
-        if (targetOpt.isDefined()) {
+        final Option<IndexedGuideGroup> iggOpt = targetOpt.flatMap(this::getTargetGroup).orElse(getSelectedGroup());
+        final boolean primaryGroupIsSelected = iggOpt.exists(igg -> igg.group().equals(env.getOrCreatePrimaryGuideGroup()));
```

Comment:
Unrelated to this PR, would you add an issue, a GitHub issue seems fine, to rename `getOrCreatePrimaryGuideGroup` to just `getPrimaryGuideGroup`?  Or just add a commit to do it now I suppose.  It is no longer the case that this method will create anything ever so it is misleading.  There is always a primary guide group now.